### PR TITLE
[CDAP-15787] Updated XMLReaderBatchSource for New Validation API

### PR DESCRIPTION
**Pull request by dli357**
Updated XMLReaderBatchSource.java and XMLReaderConfigTest.java for new validation API. Also unable to run XMLReaderBatchSourceTest due to InvalidArtifactException.

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java [hunk 1]
```diff
@@ -106,7 +115,7 @@ XMLReaderConfig getConfig() {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
-    config.validate();
+    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
     pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_XML_SCHEMA);
     if (!config.containsMacro("tableName") && !Strings.isNullOrEmpty(config.tableName)) {
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java:113]: this is a relic of the past and will not work well on Hadoop, and will not work at all in a cloud environment.

Please set the widget type for the table name and the table expiry to 'hidden', and remove mention of them from the documentation.

File: core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java [hunk 2]
```diff
@@ -317,46 +326,62 @@ String getNodePath() {
...
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(nodePath), "Node path cannot be empty.");
+      if (!containsMacro(NODE_PATH) && Strings.isNullOrEmpty(nodePath)) {
+        collector.addFailure("Node path cannot be empty.", null).withConfigProperty(NODE_PATH);
       }
 
-      if (!containsMacro("tableExpiryPeriod") && tableExpiryPeriod != null && tableExpiryPeriod < 0) {
-        throw new IllegalArgumentException("Value for Table expiry period should either be empty or greater than 0.");
+      if (!containsMacro(TABLE_EXPIRY_PERIOD) && tableExpiryPeriod != null && tableExpiryPeriod < 0) {
+        collector.addFailure(String.format("Invalid value: %d.", tableExpiryPeriod),
+            "Value for 'Table Expiry Period' should either be empty or greater than 0")
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java]: indentation (rest of PR as well)

File: core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java (hunk 1 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java:113]: Sounds good, fixed

File: core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java (hunk 2 above)
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java]: Fixed

**Review by albertshau — APPROVED** [review]

**Pull request merged**